### DISCUSSION TITLE
I have a few ideas for Sketchware Pro

### DISCUSSION THREAD

**Discussion by thisisshadow** [Sketchware-Pro/Sketchware-Pro#972] [Ideas]
I tried the latest debug version and the material design looks very good, congratulations you are working very well 🤓

I want to tell you a few features, if possible, they might really work.

**1.** After adding the components to the page, when there is an error in the component name, component editing can be added instead of deleting it and adding it from the beginning.

**2.** Instead of starting the aab compiler outside the project, when the run button is clicked, there may be options such as compile aab and compile apk for convenience.

**3.** Is the SketcwareUtil.java class really a necessary class? If not, this class can be optionally added or removed from the project.

**4.** Can asd be added for MoreBlocks, so to explain in more detail, for example, I create a private void with the block generator, but I want to do it with codes for more detail.

These are just a few features that come to my mind. If it is possible to add them, I would love to see them. :)

And while I'm writing this article, there is an error, I want to report it. 

The ID value in the View Editor is not unique, for example, there is an item named textview1, but I can make a second item textview1. An item with this id does not give the warning that it already exists, and when I edit it, the application crashes and everything I do is deleted.

Good work my friends 🖤

**Comment by j-px** [comment]
1. you can click on a Component and then you can edit it
2. that doesn't make much sense, you won't be exporting .aab files more than just a few times per day. The run button is for quick builds for testing of the project, not more.
3. see #961, a feature to remove them is planned and would be opt-in

**Reply by thisisshadow** [reply]
I understand, my friend, it's okay, but I would be very happy if the ID error I mentioned about this issue is resolved. :)

**Reply by ZylaconeEpindigo** [reply]
That ID error occurs constantly for me, if I wish it. I haven't tried with just one upper case character in the ID ( in so far as I recall, but I just tried with TextView1 when the app already had a TextView1, and it would've let me rename which would be bad karma. It maybe converting the new ID to lower case to perform the ID check. If so dropping the conversion might solve that issue. I'm old school, when we used to call computer programmers software engineers, and before we also began calling data entry specialists computer programmers. From those old archaic days of something called DOS, so I became quite accustomed to upper casing the names of variables, subroutines, functions, and objects.  I'm so glad I still can, even though that can rarely lead to an AWWWWEEE moment.

I don't recall if Sketchware crashed, seems like it didn't. I was easily able to recover the project last time by renaming the newly named. 

Anyway, it won't under those circumstances tell you of the preexisting ID.

**Comment by j-px** [comment]
btw that bug you reported isn't occurring to me:  
![Screenshot_20230910_202132_Sketchware Pro.png](https://github.com/Sketchware-Pro/Sketchware-Pro/assets/40118124/eedf117f-1113-499d-8fe5-3b3716f621de)

**Reply by j-px** [reply]
say what exact version you used, I've got the latest nightly build and it's like that

**Reply by thisisshadow** [reply]
Dude, this error occurred in version 6.4.0-rc05 and it did not occur all the time but occasionally, it was an annoying problem. I am currently using the latest test version and I do not have this error either. If I encounter this problem, I will try to share the error code. I am sorry if I disturbed you
